packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -79,57 +109,141 @@ public function getAliasName(int $domainId, string $index): string
             'index' => $indexName,
             'body' => $definition,
         ]);
+    }
 
-        $indexes->putAlias([
-            'index' => $indexName,
-            'name' => $aliasName,
-        ]);
+    /**
+     * @param int $domainId
+     * @param string $index
+     */
+    public function migrate(int $domainId, string $index)
+    {
+        $this->createIndex($domainId, $index);
+        $this->reindexFromCurrentIndexToNewIndex($domainId, $index);
+        $this->deleteCurrentIndex($this->getAliasName($domainId, $index));
+        $this->createAliasForIndex($domainId, $index);
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     * @deprecated Deleting index using this method is deprecated since SSFW 7.3, use deleteIndexesOfCurrentAlias instead
+     */
+    public function deleteIndex(int $domainId, string $index)
+    {
+        @trigger_error(
+            sprintf('Deleting index using method "%s" is deprecated since SSFW 7.3, use %s instead', __METHOD__, 'deleteIndexesOfCurrentAlias'),
+            E_USER_DEPRECATED
+        );
+
+        $indexes = $this->client->indices();
+        $indexName = $this->getIndexName($domainId, $index);
+        if ($indexes->exists(['index' => $indexName])) {
+            $indexes->delete(['index' => $indexName]);
+        }
     }
 
     /**
      * @param int $domainId
      * @param string $index
      */
-    public function deleteIndexesOfCurrentAlias(int $domainId, string $index)
+    public function reindexFromCurrentIndexToNewIndex(int $domainId, string $index)
     {
+        $indexes = $this->client->indices();
+        $indexName = $this->getVersionedIndexName($domainId, $index);
         $aliasName = $this->getAliasName($domainId, $index);
+        if ($indexes->exists(['index' => $this->getCurrentIndexName($aliasName)]) && $indexes->exists(['index' => $indexName])) {
+            $body = [
+                'source' => [
+                    'index' => $this->getCurrentIndexName($aliasName),
+                ],
+                'dest' => [
+                    'index' => $indexName,
+                ],
+            ];
+
+            $this->client->reindex([
+                'body' => $body,
+            ]);
+        }
+    }
 
+    /**
+     * @param int $domainId
+     * @param string $index
+     */
+    public function createAliasForIndex(int $domainId, string $index)
+    {
+        $indexes = $this->client->indices();
+        $aliasName = $this->getAliasName($domainId, $index);
+        $indexName = $this->getVersionedIndexName($domainId, $index);
+
+        if (!$indexes->existsAlias(['name' => $aliasName, 'index' => $indexName]) && $indexes->exists(['index' => $indexName])) {
+            $indexes->putAlias([
+                'index' => $indexName,
+                'name' => $aliasName,
+            ]);
+        }
+    }
+
+    /**
+     * @param string $aliasName
+     */
+    public function deleteCurrentIndex(string $aliasName)
```

Comment:
If the only reasonable usage is `$this->deleteCurrentIndex($this->getAliasName($domainId, $index))`, maybe the call to `getAliasName` should be inside the method.